test/new-e2e/tests/sysprobe-functional/usmtest/test_tags.ps1

```diff
@@ -0,0 +1,130 @@
+
+
+param(
+    [Parameter(Mandatory=$true)][string]$TargetHost,
+    [Parameter(Mandatory=$true)][string]$TargetPort,
+    [Parameter(Mandatory=$true)][string]$TargetPath,
+    [Parameter(Mandatory=$true)][string[]]$ExpectedClientTags,
+    [Parameter(Mandatory=$true)][string[]]$ExpectedServerTags,
+    [Parameter(Mandatory=$true)][string]$ConnExe
+)
+function stop-servicesfortest {
+    # disable process agent so that it doesn't query the connections endpoint while we're running
+    stop-service -force datadog-process-agent -ErrorAction Stop
+
+    # stop system probe to clean out any connections we're not interested in
+    stop-service -force datadog-system-probe -ErrorAction Stop
+}
+
+function make-connectionrequest {
+    param(
+        [Parameter(Mandatory=$true)] [string]$ConnExe,
+        [Parameter(Mandatory=$true)] [string]$TargetHost,
+        [Parameter(Mandatory=$false)] [string]$Port="80",
+        [Parameter(Mandatory=$false)] [string]$TargetPath="/"
+    )
+    $list = @($TargetHost, $Port, $TargetPath)
+
+    $targetpid = (start-process $ConnExe -passthru -NoNewWindow -ArgumentList $list).id    
+    return $targetpid
+}
+
+function get-connectionsendpoint {
+    $payload = ((iwr -UseBasicParsing -DisableKeepAlive http://localhost:3333/network_tracer/connections).content | ConvertFrom-Json)
+    return $payload
+}
+
+stop-servicesfortest
+
+## The test assumes some pretty specific setup. 
+
+## assumptions
+# That there is a datadog.json or app.config in the powershell directory 
+#   C:\Windows\System32\WindowsPowerShell\v1.0
+#   it sets  DD_SERVICE: powershell  DD_ENV: staging  DD_VERSION somever
+# 
+# That the IIS site has a datadog.json or web.config in the root
+#   it sets the DD_SERVICE: service1  DD_ENV: staging DD_VERSION: 1.0-prerelease
+
+## start the system probe
+start-service datadog-system-probe
+
+## just give everything a chance to settle into place.
+Start-Sleep -Seconds 5
```

Comment:
Curious, how reliable is Sleep in the test infrastructure? Historically sleep is just not consistent enough for tests because of varying loads and overhead that throw off timing.

Reply:
Not sure what you mean by "reliable".
In powershell, that sleep will certainly be 5 seconds += a millisecond or two, which is fine for this.

The agent services have a systemic problem where they report "started" to SCM before actually being fully initialized. So the choices are to write a whole lot of retry/error checking to see if the system probe is _really_ up.  Or do this.
I agree that in a perfect world, this is not ideal.